tests/loggen/socket_plugin/socket_plugin.c

```diff
@@ -308,9 +318,19 @@ active_thread_func(gpointer user_data)
 
   int fd;
   if (unix_socket_x)
-    fd = connect_unix_domain_socket(sock_type, option->target);
+    {
+      if (!option->target)
+        ERROR("please specify target parameter\n");
+
+      fd = connect_unix_domain_socket(sock_type, option->target);
+    }
   else
-    fd = connect_ip_socket(sock_type, option->target, option->port, option->use_ipv6);
+    {
+      if (!option->target || !option->port)
+        ERROR("please specify target and port parameters\n");
```

Comment:
This validation should not be duplicated, it has nothing to do with active or idle threads IMHO.

The validation is based only on command line parameters and flags (`unix_socket_x` for example), so they can be checked before initializing threads/other functionality.

`start()` might be a better place, but adding a new method to the plugin interface is also an option (you already have `get_options_list`, something like `bool validate_options(PluginOption, GError**)` looks good to me).

Reply:
I moved the check into start function